### PR TITLE
Create the NVDIMM module

### PR DESCRIPTION
**Pull request by poncovka**
Create a new storage module for the NVDIMM devices and use it in UI.

### CODE REVIEW

File: pyanaconda/modules/storage/nvdimm/nvdimm.py [hunk 1]
```diff
@@ -46,3 +54,46 @@ def process_kickstart(self, data):
...
+        """
+        namespaces = self.get_namespaces_to_use()
+        devices = self.get_devices_to_use()
+
+        return {
+            ns_info.blockdev for ns_name, ns_info in nvdimm.namespaces.items()
+            if ns_info.mode != blockdev.NVDIMMNamespaceMode.SECTOR
+            and ns_info.blockdev
+            and ns_name not in namespaces
+            and ns_info.blockdev not in devices
```
Comment (jkonecny12) [inline pyanaconda/modules/storage/nvdimm/nvdimm.py]: This implementation looks nice but we are loosing logging messages which could be important, don't we?

File: pyanaconda/ui/gui/spokes/advstorage/nvdimm.py [hunk 2]
```diff
@@ -88,67 +87,91 @@ def run(self):
...
-            self._error = None
+        try:
+            # Finish the task.
+            task_proxy.Finish()
+        except StorageConfigurationError as e:
+            # Configuration has failed, show the error.
+            self._deviceErrorLabel.set_text(str(e))
             self._conditionNotebook.set_current_page(PAGE_RESULT_ERROR)
             self._okButton.set_sensitive(True)
+
```
Comment (jkonecny12) [inline pyanaconda/ui/gui/spokes/advstorage/nvdimm.py]: Please remove this blank line.

File: pyanaconda/modules/storage/nvdimm/nvdimm.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/modules/storage/nvdimm/nvdimm.py]: Not really, the result is logged in `ignore_nvdimm_blockdevs`.

File: pyanaconda/ui/gui/spokes/advstorage/nvdimm.py (hunk 2 above)
Comment (poncovka) [reply pyanaconda/ui/gui/spokes/advstorage/nvdimm.py]: Fixed.

File: pyanaconda/modules/storage/nvdimm/nvdimm.py (hunk 1 above)
Comment (jkonecny12) [reply pyanaconda/modules/storage/nvdimm/nvdimm.py]: I see there information that the devices were added but I'm still missing reason why they are ignored when they are ignored. Am I missing something?

**Comment by poncovka** [comment]
Rebased.

File: pyanaconda/modules/storage/nvdimm/nvdimm.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/modules/storage/nvdimm/nvdimm.py]: Fixed.

**Pull request merged**